lib/api/controllers/funnelController.js

```diff
@@ -70,13 +76,52 @@ function FunnelController (kuzzle) {
     if (this.concurrentRequests < kuzzle.config.server.maxConcurrentRequests) {
       this.concurrentRequests++;
 
+      this.pendingRequests[requestObject.requestId] = requestObject;
+
       return processRequest(kuzzle, this.controllers, requestObject, context)
         .then(responseObject => callback(null, responseObject))
         .catch(err => {
           kuzzle.pluginsManager.trigger('log:error', err);
           callback(err, new ResponseObject(requestObject, err));
+
+          if (kuzzle.config.dump.handledErrors.active) {
+            setImmediate(() => {
+              var
+                lastSeen = Date.now(),
+                request,
+                errorType,
+                errorMessage;
+
+              errorType = typeof err === 'object' && err.name ? err.name : typeof err;
+
+              if (kuzzle.config.dump.handledErrors.whitelist.indexOf(errorType) > -1) {
+                request = new RequestObject({
+                  controller: 'actions',
+                  action: 'dump',
+                  body: {}
+                });
+
+                errorMessage = err.message;
+                if (errorMessage.indexOf('\n') > -1) {
+                  errorMessage = errorMessage.split('\n')[0];
+                }
+                errorMessage = errorMessage.toLowerCase().replace(/[:,'"\.\\/\s]/g, '-').replace(/[-]+/g, '-').split('-');
+                errorMessage = errorMessage.filter(value => value !== '').join('-');
+
+                request.data.body.sufix = 'handled-'.concat(errorType.toLowerCase(), '-', errorMessage);
+
+                if (!this.lastDumpedErrors[request.data.body.sufix] || this.lastDumpedErrors[request.data.body.sufix] < lastSeen - 100000) {
```

Comment:
nitpicking: can we store this `100000` number in a constant to make the code more readable, something like "MIN_TIME_BEFORE_DUMP"?